Allow property for find_inverse_reference_by to be ordered and unordered in related object

elrayle opened the pull request:
Fixes #791

Removes the check on the parent resource to determine if the property is ordered.  This was a bug as the property is expected to be defined on the child.  The shared specs were adjusted to create classes where the property is ordered in one child class, unordered in a second child class, and the property does not exist in the parent.

Fedora was the only adapter with the bug in how it found resources through the inverse relationship. Fedora stores ordered and unordered using a different pattern, so the find_inverse_reference_by method was modified to find children using both patterns.

@escowles @tpendragon This is ready for review.

elrayle commented:
All the tests pass in CircleCI.  But SimpleCov fails...

```
Coverage (99.96%) is below the expected minimum coverage (100.00%).
SimpleCov failed with exit 2/opt/circleci/.rvm/rubies/ruby-2.4.6/bin/ruby -I/home/circleci/project/gemfiles/vendor/bundle/ruby/2.4.0/gems/rspec-core-3.9.0/lib:/home/circleci/project/gemfiles/vendor/bundle/ruby/2.4.0/gems/rspec-support-3.9.0/lib /home/circleci/project/gemfiles/vendor/bundle/ruby/2.4.0/gems/rspec-core-3.9.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.r
```

I added a test to cover the changes, so I am not sure what is not covered.  Is there an analysis similar to what Coveralls does to show what line is not being tested?

escowles commented:
@elrayle When you're logged in, there is an Artifacts tab at the top of the build (https://4742-66685502-gh.circle-artifacts.com/0/coverage/index.html). I just checked it out, and the line that's not covered is the `ordered_property?` method in `lib/valkyrie/persistence/fedora/query_service.rb`

escowles commented on lib/valkyrie/persistence/fedora/query_service.rb:
```diff
@@ -139,14 +137,14 @@ def find_inverse_references_by_unordered(resource:, property:)
         content = content_with_inbound(id: resource.id)
         ids = content.graph.query([nil, ::RDF::Vocab::ORE.proxyFor, adapter.id_to_uri(resource.id)]).map(&:subject).map { |x| x.to_s.gsub(/#.*/, '') }.map { |x| adapter.uri_to_id(x) }
         ids.uniq!
-        ids.lazy.map { |id| find_by(id: id) }.select { |o| o[property].include?(resource.id) }
```
I think we still need this check. If a resource has several ordered properties, this is the only way we would know if the one we care about is pointing at the resource, instead of another one.

elrayle commented:
@escowles I removed the method that wasn't tested since it was no longer used.  I also updated the two find methods for inverse relationship.

* unordered
  * returns ids for use in avoiding duplicates in ordered processing
  * caller creates objects from the ids
* ordered
  * receives the ids from unordered and will not create any objects for those ids
  * the check of property having the target resource was added back in
  * returns objects
  * caller combines the enumerable from unordered and ordered

This seemed like the most efficient way to get the objects without duplicates.
